plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertToNonIndexedFunctionCallFix.kt

```diff
@@ -0,0 +1,76 @@
+import org.jetbrains.kotlin.idea.intentions.appendCallOrQualifiedExpression
+import org.jetbrains.kotlin.idea.intentions.collectLabeledReturnExpressions
+import org.jetbrains.kotlin.idea.intentions.setLabel
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.createSmartPointer
+import org.jetbrains.kotlin.psi.psiUtil.getQualifiedExpressionForSelector
+import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+import org.jetbrains.kotlin.psi2ir.deparenthesize
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+class ConvertToNonIndexedFunctionCallFix(
+    element: KtFunctionLiteral,
+    private val nonIndexedFunctionName: String,
+    private val labeledReturnExpressions: List<SmartPsiElementPointer<KtReturnExpression>>
+) : KotlinQuickFixAction<KtFunctionLiteral>(element) {
+    override fun getFamilyName() = KotlinBundle.message("convert.to.0", "'$nonIndexedFunctionName'")
+
+    override fun getText() = familyName
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val functionLiteral = this.element ?: return
+        val parameterList = functionLiteral.valueParameterList ?: return
+        val call = functionLiteral.getStrictParentOfType<KtCallExpression>() ?: return
+
+        labeledReturnExpressions.forEach { it.element?.setLabel(nonIndexedFunctionName) }
+        parameterList.removeParameter(0)
+        val psiFactory = KtPsiFactory(functionLiteral)
+        val callOrQualified = call.getQualifiedExpressionForSelector() ?: call
+        callOrQualified.replace(
+            psiFactory.buildExpression {
+                appendCallOrQualifiedExpression(call, nonIndexedFunctionName)
+            }
+        )
+    }
+
+    companion object : KotlinSingleIntentionActionFactory() {
+        override fun createAction(diagnostic: Diagnostic): IntentionAction? {
+            val parameter = diagnostic.psiElement as? KtParameter ?: return null
+            val parameterList = parameter.parent as? KtParameterList ?: return null
+            val parameters = parameterList.parameters
+            if (parameters.size < 2 || parameters.first() != parameter) return null
+
+            val functionLiteral = parameterList.parent as? KtFunctionLiteral ?: return null
+            val call = functionLiteral.parentCallExpression() ?: return null
+            val functionName = call.calleeExpression?.text ?: return null
+            val (nonIndexedFunctionFqName, nonIndexedFunctionName) = nonIndexedFunctions[functionName] ?: return null
+            val context = functionLiteral.analyze(BodyResolveMode.PARTIAL)
+            if (call.getResolvedCall(context)?.isCalling(nonIndexedFunctionFqName) != true) return null
+
+            val labeledReturnExpressions =
+                functionLiteral.collectLabeledReturnExpressions(functionName, context).map { it.createSmartPointer() }
```

Comment:
I propose to avoid passing data here and there. 
Instead, let's make one function that replaces labels inside a lambda without intermediate collections?

functionLiteral.replaceLambda(...) and call this function inside `invoke`

Reply:
https://github.com/JetBrains/intellij-community/blob/3a7fd755b9b8af20fb500c53b62b6ca24587c539/plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/intentions/Utils.kt#L416-L417

Don't we need to check the target function of label using BindingContext?
When resolving in the `invoke` function, the test gives the following error:
```
Resolve is not allowed under the write action for `org.jetbrains.kotlin.idea.quickfix.ConvertToNonIndexedFunctionCallFix`!
```